module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -2013,12 +2014,26 @@ protected function getItemStatusesForBiblio($id, $patron = null)
                 $duedate = null;
             }
 
+            $extraStatusInformation = [];
```

Comment:
I'm afraid this needs a bit more refactoring so that it doesn't override the choices made above (e.g. pickStatus and the related status rankings).

Reply:
I believe this is still an issue. E.g. you might want to display "At Bindery" with higher priority than the transfer, but now this overrides the choice made by the pickStatus method. I believe the correct way would be to make pickStatus handle the transfer status too (while taking `$this->statusRankings` into account) and return the extra status information in addition to the chosen status.

Reply:
I think this has now been improved, but one more question for @alexklbuckley: do we only want to include the datesent value if `$status` is set to `HoldingStatus::transit_to_date`? While I don't think it hurts anything if we pick a different status and still include the date information, it might be technically more appropriate to only provide it when it applies.

Reply:
As I mentioned before, I'd like it better if pickStatus method would return also the extra status information, but that could be done in the future. As far as I can see this is now working without side-effects.

Reply:
Thanks, @EreMaijala. Since determining the status and the extra status information requires significantly different inputs, I wonder if it would actually make more sense to create a new support method for extra status information instead of overloading pickStatus. In any case, as you say, that refactoring can be done in future as needed (e.g. if the "more status" logic gets longer and more complicated). For now, I'll merge this as-is.